### ISSUE TITLE
Support for arbitrary values in a bundle manifest

### ISSUE THREAD

**Issue by viovanov** [open-policy-agent/opa#3431]
As the author of a bundle, I'd like to store metadata in the manifest file, that travels with the bundle, and is available for querying from the runtime's [`Storage`](https://github.com/open-policy-agent/opa/blob/27c8d75d3528dc8a04e716d5455f97ee7fb537bc/storage/interface.go#L18).

The proposal is to have a key named `metadata` in the top level [`Manifest`](https://github.com/open-policy-agent/opa/blob/27c8d75d3528dc8a04e716d5455f97ee7fb537bc/bundle/bundle.go#L91) struct, that can contain an arbitrary structure.

An example use case is to store the version of the github action used to build the bundle.

**Comment by srenatus** [comment]
> github action used to build the bundle

💭 There's a github action for building bundles? ✨

**Comment by anderseknert** [comment]
Though not terribly well documented, you can currently add custom metadata claims to the .signatures.json file by providing the `--claims-file` argument, like:

```shell
opa build --signing-alg HS256 --signing-key ignored --claims-file claims.json --bundle policy/
```
I don't thing this data is available for querying though. Providing a `--claims-file` without a `--signing-key` is currently a no-op, but I suppose that we could put that data in the manifest file in that case.. or would you ever want to have custom claims in _both_?  🤔

**Comment by viovanov** [comment]
@anderseknert I'm not thinking about these metadata as claims. They should be available in the bundle even if it's not signed.
I started a PR to gather more feedback if this makes sense. Thanks!

**Comment by viovanov** [comment]
@anderseknert one more thing - I misspoke about "query" - I only meant to say that it's stored in `Storage`. I updated the description.

**Comment by anderseknert** [comment]
> I'm not thinking about these metadata as claims. They should be available in the bundle even if it's not signed.

Yeah, the suggestion was to have them added to the manifest metadata if not using signed bundles. Your PR does not include any way of actually putting the claims in the manifest file other than manually adding them there. Whether reusing `--claims-file` for this would be a good idea though, I'm not sure 😄 I was mainly just thinking out loud. It would be nice with _some_ way of including them when running `opa build`.

> I misspoke about "query" - I only meant to say that it's stored in Storage. I updated the description.

Ah, OK! That would only make them available to users of the Go library then. If we included this I'm thinking we should expose it in some form on the server too, like perhaps in [provenance](https://www.openpolicyagent.org/docs/latest/rest-api/#provenance) data, maybe even in reported [status](https://www.openpolicyagent.org/docs/latest/management-status/).

I left a couple of comments on your PR too, but since this touches on design of internals, pehaps @tsandall or @ashutosh-narkar could chime in before we proceed further.

Also, I'd love to learn more about how you would use this metadata!

**Comment by viovanov** [comment]
Makes sense @anderseknert, I agree it would be useful to be able to include these from the command line.
For being able to query, I just realized today that the info _is_ available in `data.system` as a side effect of being in the manifest. That works out great for us.

Thanks for the comments, I hope I covered everything in my latest push.

Our main use-case for this is to describe the format of package names. We make assumptions about how they're formed right now. If the bundle manifest would include a description of how these names were formed in metadata, we'd no longer need to make these assumptions.
